### PR TITLE
Re-implement config file management to use YAML

### PR DESCRIPTION
**Pull request by JasperGray**

### CODE REVIEW

File: Floofbot/Configs/BotConfigFactory.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        public static void Initialize(string filename)
+        {
+            _filename = filename;
+        }
+
+        class BotConfigParser
+        {
+            public static BotConfig ParseFromFile(string filename)
```
Comment (jkchen2) [inline Floofbot/Configs/BotConfigFactory.cs:32]: No error handling on missing file?

File: Floofbot/Handlers/RandomResponseGenerator.cs [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
+            if ((response.RequireExact && loweredMessageContent == loweredRequiredInput)
+                || loweredMessageContent.Contains(loweredRequiredInput))
+            {
+                if (val < response.Probability)
+                {
+                    return response.Response;
+                }
+                else
+                {
+                    break;
```
Comment (jkchen2) [inline Floofbot/Handlers/RandomResponseGenerator.cs:29]: Should this be `continue`?

Comment (JasperGray) [reply Floofbot/Handlers/RandomResponseGenerator.cs:29]: Oh true. It's possible that a higher-probability response that also matches comes later.

File: Floofbot/Configs/BotConfigFactory.cs (hunk 1 above)
Comment (JasperGray) [reply Floofbot/Configs/BotConfigFactory.cs:34]: Fixed

**Pull request merged**